src/vulkan/vk-command.cpp

```
@@ -506,6 +515,15 @@ void CommandRecorder::cmdUploadTextureData(const commands::UploadTextureData& cm
             // considered to be tightly packed according to the imageExtent.
 
             // Calculate the row length (in texels) from the supplied pitch (in bytes)
+            if (srLayout->rowPitch % srLayout->colPitch != 0)
+            {
+                m_device->handleMessage(
+                    DebugMessageType::Error,
+                    DebugMessageSource::Layer,
+                    "Buffer-to-texture row pitch must be a multiple of the format block size on Vulkan."
+                );
```

Cppcheck (2.21.0)

[style] The function 'shouldDeferPipelineCompilation' is never used.

(unusedFunction)